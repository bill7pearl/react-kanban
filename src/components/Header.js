import React from 'react';
import { NavLink, Route, Routes } from 'react-router-dom';
import Rockets from './Rockets';
import Missions from './missions';
import MyProfile from './myProfile';
import Img from '../Logo.png';

const Header = () => (
  <div className="header-top">
    <div className='logo'>
      <img className="img" src={Img} alt="logo" />
      <h1>Space Travelers' Hub</h1>
    </div>
    <div>
    <nav className="nav-link">
      <li><NavLink to="/">Rockets</NavLink></li>
      <li><NavLink to="/missions">Missions</NavLink></li>
      <li className='li'><NavLink to="/MyProfile">My Profile</NavLink></li>
    </nav>
    <Routes>
        <Route path="/" element={<Rockets />} />
        <Route path="/missions" element={<Missions />} />
        <Route path="/myProfile" element={<MyProfile />} />
      </Routes>
    </div>
 
  </div>
);

export default Header;