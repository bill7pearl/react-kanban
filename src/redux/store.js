import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import rocketReducer from './rockets/rockets';
import missionReducer from './missions/missions';

const store = configureStore(
  {
    reducer: {
      rockets: rocketReducer,
      missions: missionReducer,
    },
  },
  applyMiddleware(thunk),
);

export default store;
