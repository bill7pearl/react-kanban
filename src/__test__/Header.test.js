import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Header from '../components/Header';

const App = () => {
  <Provider store={store}>
    <Header />
  </Provider>;
};

describe('Header and Navigation test', () => {
  it('Tests navbar rendering with navigation', () => {
    const HeaderRenderer = renderer.create(<App />).toJSON();
    expect(HeaderRenderer).toMatchSnapshot();
  });
});
