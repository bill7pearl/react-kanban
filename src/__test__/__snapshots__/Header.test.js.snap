// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header and Navigation test Tests navbar rendering with navigation 1`] = `null`;
